### PR TITLE
Introduce logfile option and log rotation

### PR DESCRIPTION
**Pull request by itzg**
TODO:

- [x] Further testing of upstart changes
- [x] Determine best way to terminate HUP goroutine

### CODE REVIEW

File: poller.go [hunk 1]
```diff
@@ -61,19 +69,48 @@ var versionCmd = &cobra.Command{
...
+
+	if !globalFlags.NoLogfile && globalFlags.LogfileName != "" {
+		log.WithField("location", globalFlags.LogfileName).Info("Redirecting log output")
+		setLogOutput()
+
+		hupChan := make(chan os.Signal, 1)
+		signal.Notify(hupChan, os.Interrupt, syscall.SIGHUP)
+		go func() {
+			for {
+				//TODO select on a cancelling context channel, etc in order to exit
```
Comment (itzg) [inline poller.go]: @rphillips I'm not really sure I want all this living in the main package and likewise open to suggestions about terminating this goroutine (or not...just let process death happen).

Comment (rphillips) [inline poller.go]: this is fine I think...

File: poller/session_test.go [hunk 2]
```diff
@@ -432,6 +432,23 @@ func TestEleSession_PollerPrepare(t *testing.T) {
 			expectValidate:  false,
 			expectReconcile: false,
 		},
+		{
```
Comment (itzg) [inline poller/session_test.go]: @dimtruck paid some credits to coveralls...and found/fixed a bug in the process :)

**Comment by itzg** [comment]
@rphillips this is ready for one last glance: fixed the upstart service exec line.

**Comment by rphillips** [comment]
+1

**Pull request merged**